### PR TITLE
Add a Create Custom Builds tutorial

### LINKED ISSUE #3459: Create an official tutorial for custom builds
We'll soon have a tutorial for compiling application code and ol3-code, but I think basic tutorial on creating custom builds should be added as well.

We could use this a s starting point: http://boundlessgeo.com/2014/10/openlayers-custom-builds-revisited/

/cc @bartvde

### PR DESCRIPTION
**Pull request by elemoine**
As discussed in #3459 this PR adds a "Create Custom Builds" tutorial. This tutorial is based on @probins' "Custom Build Examples" in #2489.

Fixes #3459.

### CODE REVIEW

File: doc/tutorials/custom-builds.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
+---
+title: Custom builds
+layout: doc.hbs
+---
+
+# Creating custom builds
+
+OpenLayers 3 is a big library providing a lot of functionality. So it is unlikely that an application need and use all the functionality OpenLayers 3 provides. This is why creating application-specific OpenLayers 3 builds, with just the functionality your application needs, is often a good idea.
```
Comment (probins) [inline doc/tutorials/custom-builds.md]: 'application will need and use'

File: doc/tutorials/custom-builds.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+      "ol.ENABLE_VECTOR=false",
+```
+
+and re-run the build script. The build size should now be smaller.
+
+### Externs
+
+The Closure documentation explains that "externs" are for external names used in the code being compiled. The compiler includes externs for built-ins such as `document`. The `externs` directory of the OpenLayers 3 code includes files for all those used in some part of the library. For example, if you use Bing Maps, you should include the Bing externs file in the `externs` section of the config file.
+
+`oli.js` and `olx.js` are externs files for the OpenLayers 3 API. For examples `olx.js` includes extern defintions for OpenLayers 3's constructor options. You should always use these two files as externs when creating custom builds.
```
Comment (probins) [inline doc/tutorials/custom-builds.md]: definitions

File: doc/tutorials/custom-builds.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    "compilation_level": "ADVANCED_OPTIMIZATIONS",
+    "use_types_for_optimization": true,
+    "manage_closure_dependencies": true
+  }
+}
+```
+
+The exports are given here in the order in which they occur in the `heatmaps-earthquakes` example's JavaScript code. In this example we not only use the `ol.` functions and constructors, but also `prototype` methods where the `ol` namespace is not directly used. In the code, we have for example `vector.getSource().on()`. This means we are using the `getSource` method of `layer.Heatmap` and the `on` method of `source.KML`, so this is what has to be exported. Similarly, `event.feature.get()` means we are using the `feature` property of `source.VectorEvent` and the `get` method of `Feature`. If any of these names are left out, the compile will complete successfully, but the missing names will be obfuscated and you will get a 'property undefined' error when you try and run the script.
+
+As this example uses a vector layer, which has to be enabled by removing `"ol.ENABLE_VECTOR=false"` in the `define` section of the config.
```
Comment (probins) [inline doc/tutorials/custom-builds.md]: either 'This example..., which' or 'As this..., this'

File: doc/tutorials/custom-builds.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
+---
+title: Custom builds
+layout: doc.hbs
+---
+
+# Creating custom builds
+
+OpenLayers 3 is a big library providing a lot of functionality. So it is unlikely that an application need and use all the functionality OpenLayers 3 provides. This is why creating application-specific OpenLayers 3 builds, with just the functionality your application needs, is often a good idea.
+
+An alternative to creating custom builds is to compile the application code and the OpenLayers 3 code. See the [Compiling Application with Closure Compiler](closure.html) tutorial for more information.
```
Comment (gberaudo) [inline doc/tutorials/custom-builds.md]: together

Comment (tschaub) [inline doc/tutorials/custom-builds.md]: Agreed.  Or it could read "An alternative ... is to compile your application code together with the library."

**Comment by elemoine** [comment]
Thank you all for your comments. I've addressed them.

**Comment by tschaub** [comment]
Very nice.  I think the comments above are good.  Please merge after addressing them.

**Pull request merged**